### PR TITLE
Some bug fixes, a few added features

### PR DESCRIPTION
**Pull request by alejandroschuler**
- shareglrm can now take an array of Y-regularizers, just like glrm
- added periodic and imbalanced hinge loss functions
- bug fixes

### CODE REVIEW

File: src/loss_and_reg.jl [hunk 1]
```diff
@@ -117,17 +141,32 @@ function avgerror(a::AbstractArray, l::ordinal_hinge)
     sum(map(ai->evaluate(l,m,ai),a))/length(a)
 end
 
-function avgerror(a::AbstractArray, l::hinge)
```
Comment (madeleineudell) [inline src/loss_and_reg.jl]: Why did you delete this definition?

Comment (alejandroschuler) [inline src/loss_and_reg.jl]: hinge is subsumed by imbalanced_hinge, which has its own avgerror(). Something created with hinge() will be of type imbalanced_hinge with case_weight_ratio=1.

**Comment by madeleineudell** [comment]
This looks great. Just a few changes and I'll pull:

* can you add the mathematical definition of the new functions (`periodic` and `imbalanced_hinge` to the comments preceding them, and add them to the README?
* would you mind changing the name of `imbalanced_hinge` to `unbalanced_hinge` or `weighted_hinge`?

**Comment by alejandroschuler** [comment]
No problem- weighted_hinge is much better. I'll do that in a bit.

**Comment by alejandroschuler** [comment]
Alright, the changes have been made. Check it out and let me know if everything is satisfactory.

**Comment by madeleineudell** [comment]
closes #4
